proposals/manifest_trial_tokens.md

```diff
@@ -0,0 +1,176 @@
+
+Upon manifest parsing, browser should parse `trial_tokens` like so:
+ - `trial_tokens` key must never cause installation errors, only benign
+   warnings
+ - browsers can limit the total number of parsed and/or accepted tokens
+ - browsers can limit the length of each token
+ - browsers can limit the total length of all parsed and/or accepted tokens
+ - browsers can validate token structure, content, and signature at parse time;
+   in particular, tokens should not contain control characters like new lines
+   and charaxter returns which could affect HTTP header serialization and
+   parsing
+
+#### Manifest parsing algorithm
+ 1. if `trial_tokens` is not present, return
+ 2. if `trial_tokens` is not an array or if `trial_tokens` is an empty array,
+    (optionally) log a benign warning and return
+ 3. for every element `token` of `trial_tokens`:
+    1. (optionally) if the maximum number of accepted tokens is reached,
+       skip all other tokens
+    2. if `token` is not a string, skip it and (optionally) log a benign
+       warning
+    3. if `token` is an empty string, skip it and (optionally) log a benign
+       warning
+    4. (optionally) if `token` is too long to be parsed, skip it and
+       (optionally) log a benign warning
+    5. if token has been encountered before, skip it and (optionally) log
+       a benign warning
+    6. (optionally) attempt to parse the token and validate it. If validation
+       fails, skip the token and (optionally) log a benign warning.
+       Note: browsers may disregard token expiration errors if they are not
+       sure about the correctness of system timestamp at the time of token
+       validation
+    7. append token to the collection of accepted tokens
+ 4. if at least one token is accepted in step 3, save the token collection in
+    parsed manifest under `trial_tokens` key
+
+### New Permissions
+
+No new permissions.
+
+### Manifest File Changes
+
+Add new manifest member called `trial_tokens` which is a non-empty array of
+non-empty strings representing trial tokens.
+
+## Security and Privacy
+
+### Exposed Sensitive Data
+
+This new key does not expose any sensitive data by itself, but it may activate
+security- and privacy-critical code paths in the browser itself.
+
+### Abuse Mitigations
+
+Trial tokens system already incorporate some degree of abuse mitigations:
+ 1. Browser vendor signs every token for every origin and feature.
+ 2. Browser vendor audits every trial feature similarly to a regular feature.
+ 3. Browser vendor can invalidate any token individually or all tokens for
+    a particular feature remotely. Such invalidation scheme is more powerful
+    than existing method of compiling extension id allow- and block-list for
+    a feature at compilation time.
+
+### Additional Security Considerations
+
+This new manifest key contains values which will be inserted into headers of
+local HTTP-like responses. Browsers must ensure that the provided tokens can
```

Comment:
The phrasing of this section may be overly prescriptive: Chrome's implementation of origin trials is header based, but other browsers MAY use a different approach. This suggestion attempts to strike a more neutral position. If this suggestion is adopted, additional massaging may be required to make this section read well.

```suggestion
This new manifest key contains values which will be consumed by the browser's
experimental API trial system. Browsers may need to take special care to ensure
that provided tokens do not violate the expectations of these systems.

For example, Chromium's trial system exposes tokens in headers of local
HTTP-like responses. As a result, it must ensure that the provided tokens can
```

Reply:
I applied the suggestion verbatim.